docs/commonmark.md

```diff
@@ -4,5 +4,4 @@ A basic Markdown Abstract Syntax Tree, mdast, is defined at [](https://github.co
 
 ## Deviations commonmark mdast
 
-- According to the mdast spec [list items](https://github.com/syntax-tree/mdast#listitem) may only have [flow content](https://github.com/syntax-tree/mdast#flowcontent) children. However, according to the commonmark spec, list items may also be flow or phrasing content, such as [text](https://spec.commonmark.org/0.30/#example-255). This depends on spacing between the list items in the original document. In myst-spec, we choose to follow the commonmark spec and allow `ListItem` children to be `FlowContent` or `PhrasingContent`.
```

Comment:
I don't think this is correct w.r.t the CommonMark spec:
```markdown
- one

two
```
produces [this CommonMark AST](https://spec.commonmark.org/dingus/?text=-%20one%0A%0A%20two%0A):
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE document SYSTEM "CommonMark.dtd">

<document xmlns="http://commonmark.org/xml/1.0">
  <list type="bullet" tight="true">
    <item>
      <paragraph>
        <text>one</text>
      </paragraph>
    </item>
  </list>
  <paragraph>
    <text>two</text>
  </paragraph>
</document>

```

Reply:
Isn't this saying that we default to having a paragraph instead of text directly in the list item? Which is what is created?

Reply:
Our documentation states that 
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE document SYSTEM "CommonMark.dtd">

<document xmlns="http://commonmark.org/xml/1.0">
  <list type="bullet" tight="true">
    <item>
      <text>one</text>
    </item>
  </list>
</document>
```
is permitted by the CommonMark spec. I don't see any evidence that that is the case. What I did notice is that the HTML renderer doesn't show the `<p>`, but I think that might be a rendering difference rather than a spec difference.

I noticed this because whilst the spec page shows `<li>TEXT</li>` in the inline example, [the actual examples file](https://spec.commonmark.org/0.31.2/spec.json) does not:

```json
  {
    "markdown": "- one\n\n two\n",
    "html": "<ul>\n<li>one</li>\n</ul>\n<p>two</p>\n",
    "example": 255,
    "start_line": 4190,
    "end_line": 4199,
    "section": "List items"
  }
```

Reply:
Wow... yeah, wild that the html doesn't show the `<p>` despite it being in the AST! Oof, good catch.